apidoc/lib/typescript_generator.js

```diff
@@ -0,0 +1,528 @@
+'use strict';
+
+/*
+ * Map of invalid types and their replacement
+ */
+const invalidTypeMap = {
+	'2DMatrix': 'Matrix2D',
```

Comment:
We may want to rename the SDK API's for this, either to `Matrix2D` / `Matrix3D` or `Matrix` with `type: Ti.UI.MATRIX_TYPE_2D`. Thoughts?

Reply:
+1 to rename them. @garymathews  encountered a similar issue when playing around with type constructors instead of factory functions, since any type name starting with a number is invalid.

Reply:
Which option do you prefer? How do others name it? Any official specification?

Reply:
From what i have seen, most other libraries only deal with one Matrix at a time, so its simply called Matrix as there is no need to distinguish between 2D and 3D matrices.  Since we need both I'd prefer to name them `Matrix2D` and `Matrix3D` to keep a type for each as they provide a different set of transformation functions you can apply to them.